src/micromamba/install.cpp

```diff
@@ -325,78 +312,83 @@ install_specs(const std::vector<std::string>& specs, bool create_env, int solver
 void
 parse_file_options()
 {
-    if (create_options.files.size() == 0)
-    {
+    auto& configuration = Configuration::instance();
+    auto& file_specs
+        = configuration.at("file_specs").compute_config().value<std::vector<std::string>>();
+    auto& env_name = configuration.at("env_name");
+    auto& specs = configuration.at("specs");
+    auto& channels = configuration.at("channels");
+
+    if (file_specs.size() == 0)
         return;
-    }
-    for (auto& file : create_options.files)
+
+    for (auto& file : file_specs)
     {
-        if ((ends_with(file, ".yml") || ends_with(file, ".yaml"))
-            && create_options.files.size() != 1)
+        if ((ends_with(file, ".yml") || ends_with(file, ".yaml")) && file_specs.size() != 1)
         {
             throw std::runtime_error("Can only handle 1 yaml file!");
         }
     }
 
-    for (auto& file : create_options.files)
+    for (auto& file : file_specs)
     {
         // read specs from file :)
         if (ends_with(file, ".yml") || ends_with(file, ".yaml"))
         {
-            YAML::Node config = YAML::LoadFile(file);
+            YAML::Node config;
+            try
+            {
+                config = YAML::LoadFile(file);
+            }
+            catch (YAML::Exception& e)
+            {
+                LOG_ERROR << "Error in spec file: " << file;
+                continue;
+            }
+
             if (config["channels"])
             {
-                std::vector<std::string> yaml_channels;
+                YAML::Node yaml_channels;
                 try
                 {
-                    yaml_channels = config["channels"].as<std::vector<std::string>>();
+                    yaml_channels = config["channels"];
                 }
                 catch (YAML::Exception& e)
                 {
                     throw std::runtime_error(
                         mamba::concat("Could not read 'channels' as list of strings from ", file));
                 }
 
-                for (const auto& c : yaml_channels)
-                {
-                    create_options.channels.push_back(c);
-                }
+                channels.add_rc_value(yaml_channels, file);
             }
             else
             {
-                std::cout << termcolor::yellow << "WARNING: No channels specified in " << file
-                          << termcolor::reset << std::endl;
+                LOG_DEBUG << "No 'channels' specified in file: " << file;
```

Comment:
Do we have a global warning if no channels are specified?

Reply:
I don't think so. I will add one